src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -165,7 +175,7 @@ private async Task<bool> IsEnabledAsync<TContext>(string feature, TContext appCo
 
                         //
                         // IContextualFeatureFilter
```

Comment:
`if (useAppContext && ContextualFeatureFilterEvaluator.IsContextualFilter(filter, typeof(TContext)))`

At this point we should already know whether the filter is a contextual filter or not by whether filter was populated with or without an app context.